lib/forspell/loaders/ruby.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'yard'
+
+module Forspell::Loaders
+  class Ruby < Base
+    def initialize(file: nil)
+      super
+      @parser_class = YARD::Parser::Ruby::RubyParser
+    end
+
+    def load_comments
+      @parser_class.new(@input, @file).parse
+                   .tokens.select { |token| token.first == :comment }
+    rescue YARD::Parser::ParserSyntaxError, RuntimeError => e
+      raise Forspell::Loader::ParsingError, e.message
+      # example: [:comment, "# def loader_class path\n", [85, 2356]]
+    end
+
+    def load_words
+      inputs = []
+
+      load_comments.each do |comment|
+        paragraph = inputs.find { |i| i[:end] == comment.last.first }
+        if paragraph
+          inputs.delete(paragraph)
+          paragraph[:end] = comment.last.first + 1
+          paragraph[:text] << comment[1]
+          inputs << paragraph
+        else
+          inputs << { start: comment.last.first, end: comment.last.first + 1, text: comment[1] }
+        end
+      end
```

Comment:
Also, it will probably be easier to first create (start, end, text) structures, and then join paragraphs if necessary